packages/editor/src/components/rich-text/index.js

```diff
@@ -801,29 +804,87 @@ export class RichText extends Component {
 				return false;
 			}
 
+			// Allow primitives and arrays:
+			if ( ! isPlainObject( this.props[ name ] ) ) {
+				return this.props[ name ] !== prevProps[ name ];
```

Comment:
Why are we comparing non `format_` props?

Reply:
At this point in the code, is is guaranteed that `name` starts with `format_`.

Reply:
I think this needs more commenting. I don't understand what's going on.